src/topology/extremally_disconnected.lean

```diff
@@ -0,0 +1,130 @@
+/-
+Copyright (c) 2021 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+import topology.compact_open
+import topology.stone_cech
+import topology.category.CompHaus
+
+/-!
+
+# Extremally disconnected spaces
+
+An extremally disconnected topological space is a space
+in which the closure of every open set is open.
+The `class` of extremally disconnected spaces is called `extremally_disconnected`.
+Such spaces are also called Stonean spaces.
+They are the projective objects in the category of compact Hausdorff spaces.
+
+The definition `compact_t2.projective` corresponds to the mathematical statement that
+a topological space `X` is a projective object in the category of compact Hausdorff spaces.
+
+
+Lemma `extremally_disconnected_of_projective` shows that compact Hausdorff spaces that are
+projective are extremally disconnected.  The converse is missing.
+
+# TODO
+
+Prove that a compact, Hausdorff, extremally disconnected space is a projective object in the
+category of compact Hausdorff spaces.
+
+## References
+
+Gleason, Andrew M. (1958), "Projective topological spaces",
+Illinois Journal of Mathematics, 2 (4A): 482–489,
+doi:10.1215/ijm/1255454110, MR 0121775
+-/
+
+universe variables u v w
+
+noncomputable theory
+open_locale classical
+
+variables (X : Type u) [topological_space X]
+
+open function
+
+/-- An extremally disconnected topological space is a space
+in which the closure of every open set is open. -/
+class extremally_disconnected : Prop :=
```

Comment:
There are a number of connected and separation properties in [topology.connected](https://github.com/leanprover-community/mathlib/blob/master/src/topology/connected.lean) which are defined on subsets. For consistency, should `extremally_disconnected` be defined in a similar way e.g. something like:
```
def is_extremally_disconnected (S : set X) : Prop :=
∀(U : set X), is_open U → ∃(V : set X), is_open V ∧ S ∩ closure U = S ∩ V

class extremally_disconnected_space (X : Type u) [topological_space X] : Prop :=
(is_extremally_disconnected_univ : is_extremally_disconnected (set.univ : set X))
```
?